ingestor/adx/tasks.go

```diff
@@ -215,16 +218,84 @@ func (t *ManagementCommandTask) Run(ctx context.Context) error {
 		stmt := kql.New(".execute script<|").AddUnsafe(command.Spec.Body)
 		if _, err := t.kustoCli.Mgmt(ctx, stmt); err != nil {
 			logger.Errorf("Failed to execute management command %s.%s: %v", command.Spec.Database, command.Name, err)
-			if err = t.store.UpdateStatus(ctx, command, err); err != nil {
+			if err = t.store.UpdateStatus(ctx, &command, err); err != nil {
 				logger.Errorf("Failed to update management command status: %v", err)
 			}
 		}
 
 		logger.Infof("Successfully executed management command %s.%s", command.Spec.Database, command.Name)
-		if err := t.store.UpdateStatus(ctx, command, nil); err != nil {
+		if err := t.store.UpdateStatus(ctx, &command, nil); err != nil {
 			logger.Errorf("Failed to update success status: %v", err)
 		}
 	}
 
 	return nil
 }
+
+type SummaryRuleTask struct {
+	store    storage.CRDHandler
+	kustoCli StatementExecutor
+}
+
+func NewSummaryRuleTask(store storage.CRDHandler, kustoCli StatementExecutor) *SummaryRuleTask {
+	return &SummaryRuleTask{
+		store:    store,
+		kustoCli: kustoCli,
+	}
+}
+func (t *SummaryRuleTask) Run(ctx context.Context) error {
+	summaryRules := &v1.SummaryRuleList{}
+	if err := t.store.List(ctx, summaryRules); err != nil {
+		return fmt.Errorf("failed to list summary rules: %w", err)
+	}
+	for _, rule := range summaryRules.Items {
+		if rule.Spec.Database != t.kustoCli.Database() {
+			continue
+		}
+		if rule.DeletionTimestamp != nil {
+			// rule is being deleted, we might possibly want to delete the associated
+			// destination table, but there is a potential for unintended historical
```

Comment:
Definitely don't think we should delete the table by deleting the rule.  Deleting the table should occur though separate affirmative action.